h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -625,18 +636,30 @@ public void computeImpl() {
 
         // Use closed form solution for X if quadratic loss and regularization
         _job.update(1, "Initializing X and Y matrices");   // One unit of work
-        double[/*k*/][/*features*/] yinit = initialXY(tinfo, dinfo._adaptedFrame, model, na_cnt);
-        Archetypes yt = new Archetypes(ArrayUtils.transpose(yinit), true, tinfo._catOffsets, numLevels);  // Store Y' for more efficient matrix ops (rows = features, cols = k rank)
+
+        double[/*k*/][/*features*/] yinit = initialXY(tinfo, dinfo._adaptedFrame, model, na_cnt); // on normalized A
+        // Store Y' for more efficient matrix ops (rows = features, cols = k rank)
+        Archetypes yt = new Archetypes(ArrayUtils.transpose(yinit), true, tinfo._catOffsets, numLevels);
         Archetypes ytnew = yt;
         double yreg = _parms._regularization_y.regularize(yt._archetypes);
-        if (!(_parms._init == GlrmInitialization.User && _parms._user_x != null) && hasClosedForm(na_cnt))    // Set X to closed-form solution of ALS equation if possible for better accuracy
+        // Set X to closed-form solution of ALS equation if possible for better accuracy
+        if (!(_parms._init == GlrmInitialization.User && _parms._user_x != null) && hasClosedForm(na_cnt))
           initialXClosedForm(dinfo, yt, model._output._normSub, model._output._normMul);
 
         // Compute initial objective function
         _job.update(1, "Computing initial objective function");   // One unit of work
-        boolean regX = _parms._regularization_x != GlrmRegularizer.None && _parms._gamma_x != 0;  // Assume regularization on initial X is finite, else objective can be NaN if \gamma_x = 0
-        ObjCalc objtsk = new ObjCalc(_parms, yt, _ncolA, _ncolX, dinfo._cats, model._output._normSub,
-                                     model._output._normMul, model._output._lossFunc, weightId, regX);
+        // Assume regularization on initial X is finite, else objective can be NaN if \gamma_x = 0
+        boolean regX = _parms._regularization_x != GlrmRegularizer.None && _parms._gamma_x != 0;
+
+        // perform normalization on matrix A if needed.
```

Comment:
Tomas:

The runtime number to the left is the one without any changes.  The numbers on the right is the one with changes to standardization and memory optimization.  I am running codes to tests run time only with standardization and then only with memory optimization to see which part actually helps most.  Please stay tuned for the results.  Both versions take about 13 iterations before ending.

I am going to check int the benchmarking script next.  You can actually run it on your own if you want.  They are called: runit_glrm_pubdev_3454_NOFEATURE_20runs.R, runit_glrm_pubdev_3454_NOFEATURE.R


Thanks,
Wendy